src/main/java/org/candlepin/pki/certs/AnonymousCertificateGenerator.java

```diff
@@ -0,0 +1,382 @@
+import java.security.cert.X509Certificate;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+
+/**
+ * The ContentAccessManager provides management operations for organization and consumer level
+ * content access modes.
+ */
+public class AnonymousCertificateGenerator {
+    private static final Logger log = LoggerFactory.getLogger(AnonymousCertificateGenerator.class);
+    private static final String BASIC_ENTITLEMENT_TYPE = "basic";
+
+    private final CertificateSerialCurator serialCurator;
+    private final X509V3ExtensionUtil v3extensionUtil;
+    private final AnonymousCloudConsumerCurator anonCloudConsumerCurator;
+    private final AnonymousContentAccessCertificateCurator anonContentAccessCertCurator;
+    private final ProductServiceAdapter prodAdapter;
+    private final AnonymousCertContentCache contentCache;
+    private final PemEncoder pemEncoder;
+    private final KeyPairGenerator keyPairGenerator;
+    private final Signer signer;
+    private final Provider<X509CertificateBuilder> certificateBuilder;
+
+    @Inject
+    public AnonymousCertificateGenerator(
+        X509V3ExtensionUtil v3extensionUtil,
+        CertificateSerialCurator serialCurator,
+        AnonymousCloudConsumerCurator anonCloudConsumerCurator,
+        AnonymousContentAccessCertificateCurator anonContentAccessCertCurator,
+        ProductServiceAdapter prodAdapter,
+        AnonymousCertContentCache contentCache,
+        PemEncoder pemEncoder,
+        KeyPairGenerator keyPairGenerator,
+        Signer signer,
+        Provider<X509CertificateBuilder> certificateBuilder) {
+
+        this.serialCurator = Objects.requireNonNull(serialCurator);
+        this.v3extensionUtil = Objects.requireNonNull(v3extensionUtil);
+        this.anonCloudConsumerCurator = Objects.requireNonNull(anonCloudConsumerCurator);
+        this.anonContentAccessCertCurator = Objects.requireNonNull(anonContentAccessCertCurator);
+        this.prodAdapter = Objects.requireNonNull(prodAdapter);
+        this.contentCache = Objects.requireNonNull(contentCache);
+        this.pemEncoder = Objects.requireNonNull(pemEncoder);
+        this.keyPairGenerator = Objects.requireNonNull(keyPairGenerator);
+        this.signer = Objects.requireNonNull(signer);
+        this.certificateBuilder = Objects.requireNonNull(certificateBuilder);
+    }
+
+    /**
+     * Creates a new certificate. Certificate content and payload is cached and reused.
+     *
+     * @param consumer An anonymous consumer for which to create a certificate.
+     * @return Anonymous content access certificate
+     */
+    public AnonymousContentAccessCertificate createCertificate(AnonymousCloudConsumer consumer) {
```

Comment:
The naming conventions chosen for the public generation op are different from generator to generator. I'm not sure if we have the ability to standardize these with an interface without getting too crazy with Java memes and stereotypes, but we can at the very least choose to normalize on something like "generate" or "createCertificate" across the lot of them.

Reply:
I don't think there's a way to unify them behind an interface as they all do something different, but I at least unified the method name to `generate`.